server/middleware/users.coffee

```diff
@@ -608,6 +611,34 @@ module.exports =
     ]
     return res.sendStatus(200)
 
+  exportData: wrap (req, res) ->
+    unless req.user
+      throw new errors.Unauthorized()
+    if req.user.isAdmin()
+      user = yield database.getDocFromHandle(req, User)
+      if not user
+        throw new errors.NotFound('User not found.')
+    else
+      if req.params.handle isnt req.user.id
+        throw new errors.Forbidden('Users may only export their own data.')
+      user = req.user
+    results = yield {
+      'level.sessions': LevelSession.find({creator: user.id})
+      'earned.achievements': EarnedAchievement.find({user: user.id})
+      'payments': Payment.find({recipient: user.get('_id')})
```

Comment:
Is there a difference between `user.id` and `user.get('_id')` ?

Reply:
It seemed in this case that `user.id` was a string and `user.get('_id')` was an ObjectId. I could be wrong? Using `user.id` did not work in some of these fetches until I changed it to `user.get('_id')`.

Reply:
Oh, I see. Looks good then.